### DISCUSSION TITLE
How can I force ideavim to always searchforward?

### DISCUSSION THREAD

**Discussion by red8888** [JetBrains/ideavim#605] [Q&A]
In my ideavimrc I have this:

```
" always search forward
nnoremap <expr> n 'Nn'[v:searchforward]
nnoremap <expr> N 'nN'[v:searchforward]
```

I do this in Vim so it always searches forward no matter where the initial match is. But if I add this to my ideavimrc pressing N or n does nothing (it breaks move to next match completely).

Does ideavim not support v:searchforward? Is there another way I could accomplish this with ideavim?

**Comment by ljchen1** [comment]
This seems a bug. I have created a bug report here: https://youtrack.jetbrains.com/issue/VIM-2885/vsearchforward-doest-work-for-IdeaVim
